currency-rates/internal/queue/notification.go

```diff
@@ -0,0 +1,36 @@
+package queue
+
+import (
+	"context"
+	"encoding/json"
+	"time"
+)
+
+var notificationEventID uint8 = 1;
+var notificationEventType = "sended";
+
+type notification struct {
+	Rate   float64  `json:"rate"`
+	Emails []string `json:"emails"`
+	Timestamp time.Time `json:"timestamp"` 
+	EventID uint8 `json:"eventid"` 
+	EventType string `json:"eventtype"` 
+}
+
+func (queue *Queue) Publish(ctx context.Context, rate float64, emails []string) error {
```

Comment:
This method seems to be more general, not just for specific case. Maybe it's better that Publish method will take interface argument, so it can be reused for another messages, not only subscription?

Reply:
Good point, thanks)